--- src/resmom/linux/mom_mach.c	2007-10-18 19:52:19.000000000 +0200
+++ src/resmom/linux/mom_mach.c	2007-11-01 15:44:46.000000000 +0100
@@ -1155,7 +1155,7 @@
         if (LOGLEVEL >= 2)
           {
           sprintf(log_buffer,"setting cpu time limit to %ld for job %s",
-            reslim.rlim_cur,
+            (long int) reslim.rlim_cur,
             pjob->ji_qs.ji_jobid);
 
           log_record(PBSEVENT_SYSTEM,0,id,log_buffer);
@@ -1197,7 +1197,7 @@
           if (LOGLEVEL >= 0)
             {
             sprintf(log_buffer,"cannot set file limit to %ld for job %s (value too large)",
-              reslim.rlim_cur,
+              (long int) reslim.rlim_cur,
               pjob->ji_qs.ji_jobid);
 
             log_err(0,id,log_buffer);
@@ -1213,7 +1213,7 @@
         if (setrlimit(RLIMIT_FSIZE,&reslim) < 0)
           {
           sprintf(log_buffer,"cannot set file limit to %ld for job %s (setrlimit failed - check default user limits)",
-            reslim.rlim_max,
+            (long int) reslim.rlim_max,
             pjob->ji_qs.ji_jobid);
 
           log_err(errno,id,log_buffer);
@@ -1297,7 +1297,7 @@
         if (setrlimit(RLIMIT_DATA,&reslim) < 0)
           {
          sprintf(log_buffer,"cannot set data limit to %ld for job %s (setrlimit failed w/errno=%d - check default user limits)",
-            reslim.rlim_max,
+            (long int) reslim.rlim_max,
             pjob->ji_qs.ji_jobid,
             errno);
 
@@ -1307,7 +1307,7 @@
         if (setrlimit(RLIMIT_RSS,&reslim) < 0)
           {
          sprintf(log_buffer,"cannot set RSS limit to %ld for job %s (setrlimit failed w/errno=%d - check default user limits)",
-            reslim.rlim_max,
+            (long int) reslim.rlim_max,
             pjob->ji_qs.ji_jobid,
             errno);
 
