--- process_request.c.org	2013-09-10 09:39:48.163386290 +0200
+++ process_request.c	2013-09-10 09:41:20.726592234 +0200
@@ -640,6 +640,23 @@
         log_buffer);
       }
 
+    /* HvB patch backport from 2.5 */
+    if (svr_conn[sfds].cn_authen != PBS_NET_CONN_FROM_PRIVIL)
+      {
+      sprintf(log_buffer, "request type %s from host %s rejected (connection not privileged)",
+        reqtype_to_txt(request->rq_type),
+        request->rq_host);
+
+      log_record(PBSEVENT_JOB, PBS_EVENTCLASS_JOB, id, log_buffer);
+
+      req_reject(PBSE_BADHOST, 0, request, NULL, "request not authorized");
+
+      close_client(sfds);
+
+      return;
+      }
+    /* End HvB patch backport from 2.5 */
+
     if (!tfind(svr_conn[sfds].cn_addr, &okclients))
       {
       sprintf(log_buffer, "request type %s from host %s rejected (host not authorized)",
